### ISSUE TITLE
[BUG] Ubuntu Focal (20.04) netplan config using network.managed

### ISSUE THREAD

**Issue by AmiramMaimon** [saltstack/salt#62219]
Hey, 

When using the `network.managed` state on Ubuntu Focal (20.04) version in order to configure/update an interface its edit the interfaces file instead of the netplan files.

It causes an unexpected behavior when restarting (applying) the netplan service like:
  - Removing the salt configuration
  - Failed to reload the netplan service

[network.managed example](https://docs.saltproject.io/en/latest/ref/states/all/salt.states.network.html#managing-network-interfaces), any example would behave the same --> editing the "wrong" file.

I would expect that it would edit the `/etc/netplan/*.yaml` files and not the `/etc/network/interface` file

**Versions**
Master version: `3004`
Minion version: `3004.1` running on `Ubuntu 20.04.2 LTS`

Best regards,
Amiram

**Comment by OrangeDog** [comment]
Related: #54791 and #57541

I suggest using `file.serialized` instead of the very outdated `network.managed`, along with a `cmd.run`:

```yaml
netplan apply:
  cmd.run:
    - onchanges:
      - file: /etc/netplan/*
```

**Comment by AmiramMaimon** [comment]
If that true and you are not going to update the `network.managed` status deprecate it or at least add a warning that it outdated.

**Comment by jlkaiser** [comment]
Yes, exactly what AmiramMaimon said. Too many rabbit holes before finally getting here.

**Comment by Grayfoox** [comment]
this error is still present and not mentioned in documentation

**Comment by gluf-technik** [comment]
> If that true and you are not going to update the `network.managed` status deprecate it or at least add a warning that it outdated.

The function _should_ still work on non-netplan/`networkd` minions on `G@os_family:Debian` **and** (BusyBox based?) distributions like Alpine which use `/etc/network/interfaces`, too (see ["Configuring Networking" on wiki.alpinelinux](https://wiki.alpinelinux.org/wiki/Configure_Networking)), after you've changed [`if os_family == "Debian"`](https://github.com/saltstack/salt/blob/4d4901324709e49ad81f3903e86904f3855aec5e/salt/modules/debian_ip.py#L43) to `if os_family in ["Debian", "Alpine"]`.

"Should" as I still have to test a bit also how to decide when to switch to a netplan based configuration so I don't have to bother with `networkd` (directly).

**Comment by ggiesen** [comment]
I've submitted a PR for an implementation if anyone interested would like to test.
